### PR TITLE
Native languages for tongues

### PR DESCRIPTION
**Pull request by Kush1Push1**
Old:

<details>

## About The Pull Request

If a mob (human one) speaks with a language their tongue tracks as native (for their language's anatomy), the message's text will not be changed with default language's text replaces.
Basicly, if a lizard (or other human mob) with lizard tongue speaks with draconic, their speech _in most cases_ will be without accent changes.

How it looks:
<details>

image

</details>

So yes, due the check's specifity (it checks selected language) human mobs can speak with other languages without accent if they select their native one and will use language prefixes. "That is a feature, not a bad code - I promise."

## Why It's Good For The Game

It looks a bit strange when races speak at their own language with difficult without real reasons for it. The changes removes that part and let players roleplay characters who can speak other languages at good level (with using prefixes at start of messages) without having a proper tongue's anatomy.

</details>

## About The Pull Request

If a mob (human one) speaks with a language their tongue tracks as native (for their language's anatomy), the message's text will not be changed with default language's text replaces.
Basicly, if a lizard (or other human mob) with lizard tongue speaks with draconic, their speech  will be without accent changes.

How it looks:
<details>

dreamseeker_kULTKEYTwZ

</details>

## Why It's Good For The Game

It looks a bit strange when races speak at their own language with difficult without real reasons for it. The changes remove that part.

## Changelog

:cl: SishTis
add: If a human mob speaks with a language their tongue counts as native one (like draconic for forked tongue), their speech will sound normally.
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/tongue.dm
Comment (JohnFulpWillard) [inline code/modules/surgery/organs/tongue.dm:83]: Could be a good idea to put this to tongue's handle_speech proc instead? Otherwise you have to copy paste this for every tongue that has a native language, like here alone you forgot to put it on fly tongue.

Comment (tralezab) [reply code/modules/surgery/organs/tongue.dm:83]: ^^^^^^^^^^^^
and maybe add a linter too, SHOULD_CALL_PARENT(TRUE) will make it so children of tongues that inherit handle_speech must call parent

Comment (Kush1Push1) [reply code/modules/surgery/organs/tongue.dm:83]: > Could be a good idea to put this to tongue's handle_speech proc instead? Otherwise you have to copy paste this for every tongue that has a native language, like here alone you forgot to put it on fly tongue.

~~There's a major issue with other language. For example, aliens while speaking do hiss1-6.mp3 and with this check, they will be without hissing. And other languages have other things but just replaces of te~~

I reread you wrote, I agree, wait a bit

**Comment by LemonInTheDark** [comment]
> So yes, due the check's specifity (it checks selected language) human mobs can speak with other languages without accent if they select their native one and will use language prefixes. "That is a feature, not a bad code - I promise."

I know saycode is a mess, but this is not the right way to go about it.
Your pr is meant to be reinforcing how tongues impact language, not totally reducing it if people use this bug.

I think this pr has soul, but there's gotta be a better way then invalidating the soul you're adding by living with this bad code logic. Hell I'm pretty sure I already know how

File: code/modules/surgery/organs/tongue.dm
Comment (LemonInTheDark) [inline code/modules/surgery/organs/tongue.dm:81]: speech_args is the list of arguments to the say() proc, to be exact the say() proc from living_say.dm https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living_say.dm#L95

handle_speech is triggered by the COMSIG_MOB_SAY signal/event, sent here https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living_say.dm#L197

If you notice, the language arg is set well before that signal, at https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living_say.dm#L160. It's set in such a way that respects both speech args, and the mob's default language.

You could use the speech_args list to pull the mob's actual selected language, and use that. If you look at other things that access this list, they do so used some defines.
https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/dcs/signals.dm#L561

Uncomment the one you want to use, and use it as a define. That should work fine.

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/tongue.dm
Comment (Kush1Push1) [reply code/modules/surgery/organs/tongue.dm:81]: I've actially tried. My commit (actually second one, was just force-pushed) has name of "second edition", because the first one was literally a shitcode around signals. Working, but shitcode. I don't understand signals and args, so don't mess with them (but tried at first edition).

Alright, I'll try

Comment (LemonInTheDark) [reply code/modules/surgery/organs/tongue.dm:81]: If you've got questions, feel free to ask. Bug me on discord if it needs more back and forth stuff. I go by some variety of potato in coderbus and on tgstation

Comment (Kush1Push1) [reply code/modules/surgery/organs/tongue.dm:81]: Done

Comment (Kush1Push1) [reply code/modules/surgery/organs/tongue.dm:81]: Done

**Comment by Kush1Push1** [comment]
It looks like a code I would want to do from the start, thank you for help!

**Comment by Kush1Push1** [comment]
dreamseeker_kULTKEYTwZ

**Review by LemonInTheDark — APPROVED** [review]
Technically a species change, in theory you'd need to search out maintainer approval before making this kind of pr, see https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#banned-content
But I forgot, and I'm not going to do that to you. So consider this retroactive approval, make sure to get something explicit before opening prs that touch species behavior in future.

**Comment by Kush1Push1** [comment]
> Technically a species change, in theory you'd need to search out maintainer approval before making this kind of pr, see https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#banned-content
> But I forgot, and I'm not going to do that to you. So consider this retroactive approval, make sure to get something explicit before opening prs that touch species behavior in future.

U-gh, sorry .-.
Thanks!

**Pull request merged**
